src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -158,21 +177,37 @@ private void subscribeToShadowTopics() throws InterruptedException {
                 ShadowDeltaUpdatedSubscriptionRequest shadowDeltaUpdatedSubscriptionRequest =
                         new ShadowDeltaUpdatedSubscriptionRequest();
                 shadowDeltaUpdatedSubscriptionRequest.thingName = shadowName;
-                iotShadowClient.SubscribeToShadowDeltaUpdatedEvents(shadowDeltaUpdatedSubscriptionRequest,
+                iotShadowClient.SubscribeToShadowDeltaUpdatedEvents(
+                                shadowDeltaUpdatedSubscriptionRequest,
                                 QualityOfService.AT_LEAST_ONCE,
                                 this::processCISShadow,
                                 (e) -> LOGGER.atError()
                                         .log("Error processing shadowDeltaUpdatedSubscription Response", e))
-                        .get(TIMEOUT_FOR_SUBSCRIBING_TO_TOPICS_SECONDS, TimeUnit.SECONDS);
+                        .get(mqttOperationTimeoutSeconds, TimeUnit.SECONDS);
                 LOGGER.info("Subscribed to shadow update delta topic");
 
                 GetShadowSubscriptionRequest getShadowSubscriptionRequest = new GetShadowSubscriptionRequest();
                 getShadowSubscriptionRequest.thingName = shadowName;
-                iotShadowClient.SubscribeToGetShadowAccepted(getShadowSubscriptionRequest,
-                                QualityOfService.AT_LEAST_ONCE, this::processCISShadow,
+                iotShadowClient.SubscribeToGetShadowAccepted(
+                                getShadowSubscriptionRequest,
+                                QualityOfService.AT_LEAST_ONCE,
+                                resp -> {
+                                    reportShadowReceived();
+                                    processCISShadow(resp);
+                                },
```

Comment:
store callback as a final field to prevent recreating lambdas which may lead to duplicate calls on message receipt